src/lsp/cobol_parser/grammar.mly

```diff
@@ -3354,18 +3354,14 @@ let generate_statement :=
 
 %public let unconditional_action := ~ = go_to_statement; < >
 let go_to_statement :=
- | GO; TO?; i = procedure_name;
-   { GoTo { goto_target = i } }
- | GO; TO?; il = nel_(procedure_name); DEPENDING; ON?; i = ident;
-   { GoToDepending { goto_depending_targets = il;
-                     goto_depending_on = i; } }
  | GO; TO?; %prec lowest
    { LoneGoTo }	   (* COB85; obsolete; should be sole statement of paragraph *)
- | GO; TO?; ENTRY; t = alphanum;
-   { GoToEntry { goto_entry_target = t } }
- | GO; TO?; ENTRY; il = nel_(alphanum); DEPENDING; ON?; i = ident;
-   { GoToEntryDepending { goto_entry_depending_targets = il;
-                          goto_entry_depending_on = i } }
+ | GO; TO?; il = nel_(procedure_name); i = o(DEPENDING; ON?; ident);
+   { GoTo (GoToSimple { goto_targets = il;
+                        goto_depending_on = i; } ) }
+ | GO; TO?; ENTRY; il = nel_(loc(alphanum)); i = o(DEPENDING; ON?; ident);
```

Comment:
You can use a new rule like:
```ocaml
let alphanum_literal == ~ = alphanum; <Alphanum>
```
and then `nel_(loc(alphanum_literal));` here instead.